src/common/src/array/arrow/arrow_impl.rs

```diff
@@ -124,18 +124,464 @@ impl TryFrom<&arrow_array::RecordBatch> for DataChunk {
     }
 }
 
+/// Provides the default conversion logic for RisingWave array to Arrow array with type info.
+pub trait ToArrowArrayWithTypeConvert {
+    fn to_arrow_with_type(
+        &self,
+        data_type: &arrow_schema::DataType,
+        array: &ArrayImpl,
+    ) -> Result<arrow_array::ArrayRef, ArrayError> {
+        match array {
+            ArrayImpl::Int16(array) => self.int16_to_arrow(data_type, array),
+            ArrayImpl::Int32(array) => self.int32_to_arrow(data_type, array),
+            ArrayImpl::Int64(array) => self.int64_to_arrow(data_type, array),
+            ArrayImpl::Float32(array) => self.float32_to_arrow(data_type, array),
+            ArrayImpl::Float64(array) => self.float64_to_arrow(data_type, array),
+            ArrayImpl::Utf8(array) => self.utf8_to_arrow(data_type, array),
+            ArrayImpl::Bool(array) => self.bool_to_arrow(data_type, array),
+            ArrayImpl::Decimal(array) => self.decimal_to_arrow(data_type, array),
+            ArrayImpl::Int256(array) => self.int256_to_arrow(data_type, array),
+            ArrayImpl::Date(array) => self.date_to_arrow(data_type, array),
+            ArrayImpl::Timestamp(array) => self.timestamp_to_arrow(data_type, array),
+            ArrayImpl::Timestamptz(array) => self.timestamptz_to_arrow(data_type, array),
+            ArrayImpl::Time(array) => self.time_to_arrow(data_type, array),
+            ArrayImpl::Interval(array) => self.interval_to_arrow(data_type, array),
+            ArrayImpl::Struct(array) => self.struct_to_arrow(data_type, array),
+            ArrayImpl::List(array) => self.list_to_arrow(data_type, array),
+            ArrayImpl::Bytea(array) => self.bytea_to_arrow(data_type, array),
+            ArrayImpl::Jsonb(array) => self.jsonb_to_arrow(data_type, array),
+            ArrayImpl::Serial(array) => self.serial_to_arrow(data_type, array),
+        }
+    }
+    #[inline]
+    fn int16_to_arrow(
+        &self,
+        _data_type: &arrow_schema::DataType,
+        array: &I16Array,
+    ) -> Result<arrow_array::ArrayRef, ArrayError> {
+        Ok(Arc::new(arrow_array::Int16Array::from(array)))
+    }
```

Comment:
Any reason why we provide an arrow data type but we don't use it at all in the default implementation?

Reply:
Because it's not needed in the original default implementation. But it may need for some external system, e.g. in the iceberg, we need to know the Decimal type when we want to convert a decimal array in rw to a decimal array.